### ISSUE TITLE
A Basic HTML5 Template For Any Project — SitePoint

### ISSUE THREAD

**Issue by tg-z** [tg-z/web-clip#84]
A Basic HTML5 Template For Any Project — SitePoint                                               set Restriction 

Adjust Website to Keyboard Navigation and Screen-Readers

Read the Website's Accessibility Statement

Open the Accessibility Adjustments Interface

Quick Navigation | Click ENTER to start navigating

Quick Navigation (Alt+1) ↑↓ to navigate. ESC to close. Tab to skip.

[Main Content](#)

Homepage

[Page Header](#) [Page Middle](#) [Page Footer](#)

 

![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAPAgMAAABGuH3ZAAAAAXNSR0IArs4c6QAAAAlQTFRF/2YA////AAAALA0mBgAAAAFiS0dEAIgFHUgAAAAJcEhZcwAAEnMAABJzAYwiuQcAAAAHdElNRQfcCxUTCRv8+I/EAAAAMklEQVQI12NYtWrVCoapoaERDBMYGCSghAMLkBAQABIsDkCCkQHGAouBZeGKwXpBpgAAhtMRBKRIycIAAAAASUVORK5CYII=)
[![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAAPAgMAAAAOp6AcAAAAAXNSR0IArs4c6QAAAAxQTFRFAAAA/2YAiY55////S8IQBgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wLFRMROauDVnkAAAB7SURBVBjTpc8xDoAgDEBRLuXsKToaT0FMungfMLgz4CmMdygzC1G0yGIii/EPDM0LaYWo1Z6vdtFADtWj6ctwhKNfcTE26ehm8nHhodxAApKx3iifX3vL0AM6Y09NxDJlOUAHK2o2ihkVKbcwAiZNaSb25c9/e1Zvr3UBWOasaPHYnJwAAAAASUVORK5CYII=)
](https://www.sitepoint.com/feed/ "https://www.sitepoint.com/feed/") [![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAAPAgMAAAAOp6AcAAAAAXNSR0IArs4c6QAAAAxQTFRFAAAA/2YAiY55////S8IQBgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wLFRMROauDVnkAAAB7SURBVBjTpc8xDoAgDEBRLuXsKToaT0FMungfMLgz4CmMdygzC1G0yGIii/EPDM0LaYWo1Z6vdtFADtWj6ctwhKNfcTE26ehm8nHhodxAApKx3iifX3vL0AM6Y09NxDJlOUAHK2o2ihkVKbcwAiZNaSb25c9/e1Zvr3UBWOasaPHYnJwAAAAASUVORK5CYII=)
](https://www.sitepoint.com/a-basic-html5-template/feed/ "https://www.sitepoint.com/a-basic-html5-template/feed/") [![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAAPAgMAAAAOp6AcAAAAAXNSR0IArs4c6QAAAAxQTFRFAAAA/2YAiY55////S8IQBgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9wLFRMROauDVnkAAAB7SURBVBjTpc8xDoAgDEBRLuXsKToaT0FMungfMLgz4CmMdygzC1G0yGIii/EPDM0LaYWo1Z6vdtFADtWj6ctwhKNfcTE26ehm8nHhodxAApKx3iifX3vL0AM6Y09NxDJlOUAHK2o2ihkVKbcwAiZNaSb25c9/e1Zvr3UBWOasaPHYnJwAAAAASUVORK5CYII=)
](https://www.sitepoint.com/xmlrpc.php?rsd "https://www.sitepoint.com/xmlrpc.php?rsd")

*   [Blog](/blog)
*   [Community](/community)
*   [Library](/premium/library/)

*   [Login](/premium/sign-in)
[Join Premium](/premium/products/Z2lkOi8vbGVhcm5hYmxlL1Byb2R1Y3QvMzA3?ref_source=sitepoint&ref_medium=topnav&ref_campaign=a-basic-html5-template&ref_content=html-css)

1.  [html-css](https://www.sitepoint.com/html-css/)
2.  August 06, 2018
3.  By [Louis Lazaris](https://www.sitepoint.com/author/louis-lazaris1/)

A Basic HTML5 Template For Any Project
======================================

_This article, updated in 2018, was originally based on a chapter from [HTML5 & CSS3 for the Real World](https://www.sitepoint.com/premium/books/html5-css3-for-the-real-world-2nd-edition), by Alexis Goldstein, Louis Lazaris and Estelle Weyl._

**As you learn HTML5 and add new techniques to your toolbox, you’re likely going to want to build yourself a boilerplate from which you can begin all your HTML5-based projects. We encourage this, and you may also consider using one of the many online sources that provide a basic HTML5 starting point for you.**

In this article, we’ll look at how to get started with this. Let’s start simple, with a bare-bones HTML5 page:

```markup
<!doctype html>

<html lang="en">
<head>
  <meta charset="utf-8">

  <title>The HTML5 Herald</title>
  <meta name="description" content="The HTML5 Herald">
  <meta name="author" content="SitePoint">

  <link rel="stylesheet" href="css/styles.css?v=1.0">

</head>

<body>
  <script src="js/scripts.js"></script>
</body>
</html>
```

With that basic template in place, let’s now examine some of the significant parts of the markup and how these might differ from how HTML was written prior to HTML5.

The Doctype
-----------

First, we have the Document Type Declaration, or _doctype_. This is simply a way to tell the browser — or any other parser — what type of document it’s looking at. In the case of HTML files, it means the specific version and flavor of HTML. The doctype should always be the first item at the top of any HTML file. Many years ago, the doctype declaration was an ugly and hard-to-remember mess. For XHTML 1.0 Strict:

```markup
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
```

And for HTML4 Transitional:

```markup
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
```

Although that long string of text at the top of our documents hasn’t really hurt us (other than forcing our sites’ viewers to download a few extra bytes), HTML5 has done away with that indecipherable eyesore. Now all you need is this:

```markup
<!doctype html>
```

Simple, and to the point. The doctype can be written in uppercase, lowercase, or mixed case. You’ll notice that the “5” is conspicuously missing from the declaration. Although the current iteration of web markup is known as “HTML5,” it really is just an evolution of previous HTML standards — and future specifications will simply be a development of what we have today.

Because browsers are usually required to support all existing content on the Web, there’s no reliance on the doctype to tell them which features should be supported in a given document. In other words, the doctype alone is not going to make your pages HTML5-compliant. It’s really up to the browser to do this. In fact, you can use one of those two older doctypes with new HTML5 elements on the page and the page will render the same as it would if you used the new doctype.

The `html` Element
------------------

Next up in any HTML document is the `html` element, which has not changed significantly with HTML5. In our example, we’ve included the `lang` attribute with a value of `en`, which specifies that the document is in English. In XHTML-based syntax, you’d be required to include an `xmlns` attribute. In HTML5, this is no longer needed, and even the `lang` attribute is unnecessary for the document to validate or function correctly.

So here’s what we have so far, including the closing `</html>` tag:

```markup
<!doctype html>
<html lang="en">

</html>
```

The `head` Element
------------------

The next part of our page is the `<head>` section. The first line inside the `head` is the one that defines the character encoding for the document. This is another element that’s been simplified since XHTML and HTML4, and is an optional feature, but recommended. In the past, you may have written it like this:

```markup
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
```

HTML5 improves on this by reducing the character encoding `<meta>` tag to the bare minimum:

```markup
<meta charset="utf-8">
```

In nearly all cases, `utf-8` is the value you’ll be using in your documents. A full explanation of character encoding is beyond the scope of this article, and it probably won’t be that interesting to you, either. Nonetheless, if you want to delve a little deeper, you can read up on the topic on [W3C](http://www.w3.org/html/wg/drafts/html/master/infrastructure.html#encoding-terminology) or [WHATWG](https://html.spec.whatwg.org/multipage/infrastructure.html#encoding-terminology).

_Note: to ensure that all browsers read the character encoding correctly, the entire character encoding declaration must be included somewhere within the first 512 characters of your document. It should also appear before any content-based elements (like the `<title>` element that follows it in our example site)_.

There’s much more we could write about this subject, but we want to keep you awake — so we’ll spare you those details! For now, we’re content to accept this simplified declaration and move on to the next part of our document:

```markup
<title>The HTML5 Herald</title>
<meta name="description" content="The HTML5 Herald">
<meta name="author" content="SitePoint">

<link rel="stylesheet" href="css/styles.css?v=1.0">
```

In these lines, HTML5 barely differs from previous syntaxes. The page title (the only mandatory element inside the `head`) is declared the same as it always was, and the meta tags we’ve included are merely optional examples to indicate where these would be placed; you could put [as many valid `meta` elements](https://html.spec.whatwg.org/multipage/semantics.html#the-meta-element) here as you like.

The key part of this chunk of markup is the stylesheet, which is included using the customary `link` element. There are no required attributes for `link` other than `href` and `rel`. The `type` attribute (which was common in older versions of HTML) is not necessary, nor was it ever needed to indicate the content type of the stylesheet.

Leveling the Playing Field
--------------------------

When HTML5 was introduced, it included a number of new elements, such as `article` and `section`. You might think this would be a major problem for older browser support for unrecognized elements, but you’d be wrong. This is because the majority of browsers don’t actually care what tags you use. If you had an HTML document with a `recipe` tag (or even a `ziggy` tag) in it, and your CSS attached some styles to that element, nearly every browser would proceed as if this were totally normal, applying your styling without complaint.

Of course, such a hypothetical document would fail to validate and may have accessibility problems, but it would render correctly in almost all browsers — the exception being old versions of Internet Explorer (IE). Prior to version 9, IE prevented unrecognized elements from receiving styling. These mystery elements were seen by the rendering engine as “unknown elements,” so you were unable to change the way they looked or behaved. This includes not only our imagined elements, but also any elements that had yet to be defined at the time those browser versions were developed. That means (you guessed it) the new HTML5 elements.

The good news is that, these days, usage of IE has dropped right off, with IE11 having fallen to around 2.7% global usage (as of 2018), and versions prior to that virtually having dropped off the map. (You can view stats on browser usage and support for HTML5 features in heneral on the [Can i use](https://caniuse.com/#search=html5) site.)

If you _really_ need to support ancient browsers, though, you can still use the trusty [HTML5 Shiv](http://ejohn.org/blog/html5-shiv/), a very simple piece of JavaScript originally developed by John Resig. Inspired by an idea by Sjoerd Visscher, it made the new HTML5 elements styleable in older versions of IE. Really, though, this shouldn’t be needed now. As indicated by [Can i use](https://caniuse.com/#feat=html5semantic), HTML5 elements are supported across all modern browsers and even even their most recent older versions. (Click the “Show all” option to see all browser versions.) The one exception is that some browsers don’t recognize the newer `main` element. However, for those browsers you can still use this element, as long as you add approraite styling (such as setting it to be a block element.)

The Rest is History
-------------------

Looking at the rest of our starting template, we have the usual `body` element along with its closing tag and the closing `html` tag. We also have a reference to a JavaScript file inside a `script` element.

Much like the `link` tag discussed earlier, the `<script>` tag does not require that you declare a `type` attribute. If you ever wrote XHTML, you might remember your `script` tags looking like this:

```markup
<script src="js/scripts.js" type="text/javascript"></script>
```

Since JavaScript is, for all practical purposes, the only real scripting language used on the Web, and since all browsers will assume that you’re using JavaScript even when you don’t explicitly declare that fact, the `type` attribute is unnecessary in HTML5 documents:

```markup
<script src="js/scripts.js"></script>
```

We’ve put the `script` element at the bottom of our page to conform to best practices for embedding JavaScript. This has to do with the page-loading speed; when a browser encounters a script, it will pause downloading and rendering the rest of the page while it parses the script. This results in the page appearing to load much more slowly when large scripts are included at the top of the page before any content. It’s why most scripts should be placed at the very bottom of the page, so that they’ll only be parsed after the rest of the page has loaded.

In some cases, however, (such as with the HTML5 shiv) the script may _need_ to be placed in the head of your document, because you want it to take effect before the browser starts rendering the page.

Next Steps
----------

One way to take your HTML5 to the next level is to try out the [HTML5 Boilerplate](https://html5boilerplate.com/). This regularly updated resource provides a handy starting point for your projects, containing all the latest best practices established by hundreds of the world’s best programmers. It’s worth downloading and checking out even if you just want to pick through the code and look at how certain elements are being used these days, such as the various meta elements found in the document’s head.

Another way to take your website or web app development to the next level is to try one of the modern frameworks that are in wide use today. For example, check out SitePoint’s extensive resources on [React](https://www.sitepoint.com/learn/react/) and [Angular](https://www.sitepoint.com/learn/angular/).

 [![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2014/12/1419330230lazaris-headshot-500-96x96.jpg)](https://www.sitepoint.com/author/louis-lazaris1/) 

Meet the author

[Louis Lazaris](https://www.sitepoint.com/author/louis-lazaris1/)[](http://twitter.com/ImpressiveWebs)[](https://plus.google.com/107512330611517129044/)

Louis has been involved in the web dev industry since 2000, and is a co-editor of the HTML/CSS content on SitePoint. He blogs at [Impressive Webs](https://www.impressivewebs.com) and curates [Web Tools Weekly](http://webtoolsweekly.com/), a newsletter for front-end developers primarily focused on tools.

1.  [javascript](https://www.sitepoint.com/javascript/)
2.  August 29, 2019
3.  By [Akshay Kadam](https://www.sitepoint.com/author/akadam/)

How to Redesign Unsplash Using Styled Components
================================================

**Writing future-proof CSS is hard. Conflicting classnames, specificity issues, and so on, come up when you have to write and maintain thousands of lines of CSS. To get rid of the aforementioned issues, [Styled Components](https://www.styled-components.com) was created.**

Styled Components makes it easy to write your CSS in JS and makes sure there are no conflicting classnames or specificity issues with multiple other benefits. This makes writing CSS a joy.

In this tutorial, we’ll explore what CSS in JS is, the pros and cons of `styled-components`, and finally, we’ll redesign Unsplash using Styled Components. After completing this tutorial, you should be able to quickly get up and running with Styled Components.

_Note: Styled Components was specifically built with React in mind, so you have to be using React to use Styled Components._

Prerequisites
-------------

For this tutorial, you need a basic knowledge of [React](https://reactjs.org).

Throughout the course of this tutorial we’ll be using `yarn`. If you don’t have `yarn` already installed, then install it from [here](https://yarnpkg.com/en/docs/install).

To make sure we’re on the same page, these are the versions used in this tutorial:

*   Node 12.6.0
*   npx 6.4.1
*   yarn 1.17.3

Evolution of CSS
----------------

Before CSS-in-JS was created, the most common way to style web apps was to write CSS in a separate file and link it from the HTML.

But this caused trouble in big teams. Everyone has their own way of writing CSS. This caused specificity issues and led to everyone using `!important`.

Then came Sass. Sass is an extension of CSS that allows us to use things like variables, nested rules, inline imports and more. It also helps to keep things organized and allows us to create stylesheets faster.

Even though Sass might be thought of as an improvement over CSS, it arguably causes more harm than good without certain systems put in place.

Later, BEM came in. BEM is a methodology that lets us reduce specificity issues by making us write unique classnames. BEM does solve the specificity problem, but it makes the HTML more verbose. Classnames can become unnecessarily long, and it’s hard to come up with unique classnames when you have a huge web app.

After that, CSS Modules were born. CSS Modules solved what neither Sass nor BEM could — the problem of unique classnames — by tooling rather than relying on the name given by a developer, which in turn solved specificity issues. CSS Modules gained a huge popularity in the React ecosystem, paving the way for projects like [glamor](https://github.com/threepointone/glamor).

The only problem with all these new solutions was that developers were made to learn new syntaxes. What if we could write CSS exactly how we write it in a `.css` file but in JS? And thus `styled-components` came into existence.

Styled Components uses [Template Literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals), an ES6 feature. Template literals are string literals allowing embedded expressions. They allow for multi-line strings and string interpolation features with them.

The main selling point of Styled Components is that it allows us to write exact CSS in JS.

Styled Components has a lot of benefits. Some of the pros and cons of Styled Components are listed below.

### Pros

There are lots of advantages to using Styled Components.

1.  **Injecting Critical CSS into the DOM**
    
    Styled Components only injects critical CSS on the page. This means users only download CSS needed for that particular page and nothing else. This loads the web page faster.
    
2.  **Smaller CSS bundle per page**
    
    As it only injects styles that are used in the components on the page, bundle size is considerably smaller. You only load the CSS you need, instead of excessive stylesheets, normalizers, responsiveness, etc.
    
3.  **Automatic Vendor Prefixing**
    
    Styled Components allows you to write your CSS and it automatically vendor prefixes according to the latest standard.
    
4.  **Remove unused CSS**
    
    With Styled Components, it’s easier to remove unused CSS or dead code, as the styles are colocated with the component. This also impacts on reducing bundle size.
    
5.  **Theming is easy**
    
    Styled Components makes it really easy to theme a React applications. You can even have multiple themes in your applications and yet easily maintain them.
    
6.  **Reduces the number of HTTP requests**
    
    Since there are no CSS files for resets, normalizers, and responsiveness, the number of HTTP requests are considerably reduced.
    
7.  **Unique Classnames**
    
    Styled Components generates unique classnames every time a build step takes place. This allows avoiding naming collisions or specificity issues. No more having global conflicts and being forced to resolve them with `!important` tags.
    
8.  **Maintenance is easy**
    
    Styled Components allows you to colocate styles with the component. This allows for painless maintenance. You know exactly which style is affecting your component, unlike in a big CSS file.
    

### Cons

Of course, nothing’s perfect. Let’s look at some downsides associated with Styled Components.

1.  **Unable to Cache Stylesheets**
    
    Generally, a web browser caches `.css` files when a user visits a website for the next visit, so it doesn’t have to download the same `.css` file again. But with `styled-components`, the styles are loaded in the DOM using the `<style>` tag. Thus, they can’t be cached and every time user has to request styles when they visit your website.
    
2.  **React specific**
    
    Styled Components was made with React in mind. Thus, it’s React specific. If you use any other framework, then you can’t use Styled Components.
    
    However, there’s an alternative very similar to `styled-components` known as [emotion](https://emotion.sh) which is framework agnostic.
    

Getting Practical
-----------------

Now that we know the pros and cons of Styled Components, let’s start using it.

Go ahead and scaffold a new React project using `create-react-app`. To scaffold it, we’ll be using `npx`, which lets you download and run a package temporarily without polluting the global namespace.

Type the following in the terminal:

```bash
$ npx create-react-app unsplash-styled-components

```

Then go inside the `unsplash-styled-components` directory and install the following packages needed for the purpose of this tutorial, namely `styled-components` and `react-medium-image-zoom`:

```bash
$ cd unsplash-styled-components
$ yarn add styled-components react-medium-image-zoom

```

The `styled-components` package allows us to write actual CSS in JS while the `react-medium-image-zoom` package allows us to zoom in on an image [Medium-style](https://medium.com).

Now, once the installation is complete, delete unneeded files like `App.css`, `App.test.js`, `index.css` and `logo.svg` from the `src/` directory.

Go ahead and remove `import './index.css';` from `index.js`, as it’s no longer needed.

Replace the contents of `App.js` with the following:

```javascript
import React from 'react'

const App = () => <h1>Hello React</h1>

export default App

```

Try running the development server now by typing `yarn start` in the terminal.

You should see `Hello React` printed on the screen like this:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566955982hello-react.png)

Styled Components uses tagged [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) to style your component.

Let’s suppose you want to style the `h1` as white text on a pink background by using Styled Components. Here’s how you would do it:

```javascript
import React from 'react'
import styled from 'styled-components'

const Header = styled.h1`
    background-color: pink;
    color: white;
`

const App = () => <Header>Hello React</Header>

export default App

```

It should now look like this:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956034hello-react-using-styled-components.png)

We just created our first Styled Component `Header`. The `Header` component is assigned to `styled.h1`. The default export `styled` coming from `styled-components` has basic HTML primitives like `div`, `span`, `h1`, `h2`, etc.

Inside of `styled.h1` we write the actual CSS. Notice how we’re writing CSS rather than making style objects.

Furthermore, the `Header` component gets replaced in the browser with an `h1` tag and a unique classname. A `style` tag gets inserted into the DOM with the same unique classname. This is the basics of how Styled Components actually work.

Our styles are in the same file as our component. This means our component is co-located with styles. Thus, it’s easier to remove styles associated with a particular component, thereby eliminating dead code.

Traditionally in CSS, it’s hard to reason about, as we have to find a particular element’s `classname` or `id` or specific `selector`, then remove them one by one. It’s easy in a small project, but it gets hard as the project complexity increases. With Styled Components, no matter the complexity of the project, it’s easier to add, edit and remove styles, as they’re colocated with the component.

Theming in Styled Components
----------------------------

Let’s add theming to our project.

Create a `utils/` folder in the `src/` directory. Then create a `themes.js` file in the `utils/` folder. Paste the following in `themes.js`:

```javascript
export const themes = {
  common: {
    fontFamily:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
  },
  light: {
    type: 'light',
    bgColor: 'white',
    color: '#3E4C59',
    searchBgColor: '#E4E7EB',
    boxShadow: '0.8rem 0.8rem 1.5rem gray',
    categoryColor: '#999',
    categoryHoverColor: '#333',
  },
  dark: {
    type: 'dark',
    bgColor: '#1F2933',
    color: '#E4E7EB',
    searchBgColor: '#E4E7EB',
    boxShadow: '0.4rem 0.4rem 1.5rem #111111',
    categoryColor: '#CBD2D9',
    categoryHoverColor: '#9AA5B1',
  },
}

```

The `themes.js` file contains an object named `themes` with three properties: `common`, `light` and `dark`.

The `common` property contains common styles across themes, the `light` property contains styles for the light theme, and the `dark` property contains styles for the dark theme.

Now change `App.js` as follows:

```javascript
import React, { useState } from 'react'
import styled, { ThemeProvider } from 'styled-components'
import { themes } from './utils/themes'

const lightTheme = () => ({
  ...themes['common'],
  ...themes['light'],
})

const darkTheme = () => ({
  ...themes['common'],
  ...themes['dark'],
})

const Header = styled.h1`
    background-color: pink;
    color: white;
`

const App = () => {
  const [theme, setTheme] = useState(lightTheme())
  const setDarkTheme = () => setTheme(darkTheme())
  const setLightTheme = () => setTheme(lightTheme())
    return (
        <ThemeProvider theme={theme}>
            <>
                <Header>Hello React</Header>
                <button onClick={setLightTheme}>Make it light</button>
                <button onClick={setDarkTheme}>Make it dark</button>
            </>
        </ThemeProvider>
    )
}

export default App

```

Firstly, we’ve imported the `useState` hook from `react`, `ThemeProvider` from `styled-components` and `themes` from `./utils/themes`.

Then we’ve created two arrow functions, `lightTheme` and `darkTheme`.

The `lightTheme` function returns an object combining the `common` and `light` properties from the imported `themes` variable.

So basically, the `lightTheme` function after evaluation is this:

```javascript
const lightTheme = () => ({
    fontFamily:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    type: 'light',
    bgColor: 'white',
    color: '#3E4C59',
    searchBgColor: '#E4E7EB',
    boxShadow: '0.8rem 0.8rem 1.5rem gray',
    categoryColor: '#999',
    categoryHoverColor: '#333',
})

```

The `darkTheme` function returns an object combining the `common` and `dark` properties from the imported `themes` variable.

Thus, the `darkTheme` function after evaluation looks like this:

```javascript
const darkTheme = () => ({
    fontFamily:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    type: 'dark',
    bgColor: '#1F2933',
    color: '#E4E7EB',
    searchBgColor: '#E4E7EB',
    boxShadow: '0.4rem 0.4rem 1.5rem #111111',
    categoryColor: '#CBD2D9',
    categoryHoverColor: '#9AA5B1',
})

```

Later, in the `App` component, we use the `useState` React hook:

```javascript
const App = () => {
  const [theme, setTheme] = useState(lightTheme())
  const setDarkTheme = () => setTheme(darkTheme())
  const setLightTheme = () => setTheme(lightTheme())
    return (
        <ThemeProvider theme={theme}>
            <>
                <Header>Hello React</Header>
                <button onClick={setLightTheme}>Make it light</button>
                <button onClick={setDarkTheme}>Make it dark</button>
            </>
        </ThemeProvider>
    )
}

```

`useState` takes in the default state, which we set to `lightTheme()`. If you set it to `darkTheme()`, then by default the dark theme will be displayed. `useState` returns two variables, which we named `theme` and `setTheme`. You can name them anything.

`theme` contains the current state — that is, `lightTheme()` — and `setTheme` is a function used to set a new state.

Then we have the `setDarktheme` function, which sets a dark them by calling `setTheme` and passing it the object returned by the function `darkTheme()`. The `setLightTheme` function works in a similar way.

Finally, we have two additional buttons to change themes — one to change it to `light` and other to `dark`.

But for this to work, we need to change our `Header` component using themes. So change it to the following:

```javascript
const Header = styled.h1`
    background-color: ${props => props.theme.bgColor};
    color: ${props => props.theme.color};
`

```

Now the light theme looks like this:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956066selected-light-theme.png)

If you try clicking on the `Dark Theme` button, it will show the dark theme as follows:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956096selected-dark-theme.png)

Styled Components has a feature which lets you change your component styles dynamically using a prop. The `background-color` variable uses the prop `theme.bgColor`, while `color` uses the prop `theme.color`.

The `theme` variable mentioned here comes from `ThemeProvider`. The prop provided to `ThemeProvider` should be named `theme`. Otherwise it won’t work.

Now try clicking on the buttons to see themes changing.

The themes change, but it’s confusing which button to click. Let’s add a conditional that shows the light theme button only when the dark theme is shown, and vice versa.

Change the buttons in `App.js` as follows:

```javascript
{theme.type === 'dark' && <button onClick={setLightTheme}>Make it light</button>}
{theme.type === 'light' && <button onClick={setDarkTheme}>Make it dark</button>}

```

This conditional makes sure to only show the relevant button depending on the theme selected.

Only shows `light` button on `dark` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956131show-light-button-on-dark-theme.png)

Only shows `dark` button on `light` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956165show-dark-button-on-light-theme.png)

Go ahead and download the [assets/](https://github.com/deadcoder0904/unsplash-styled-components/tree/master/src/assets) folder and paste it into the `src/` directory.

This will ensure we have all the `images` and `icons` we need in this project.

Now let’s change the buttons to icons. We’ll make the light button as a sun icon and dark button as a moon icon.

Import the icons into `App.js`:

```javascript
import { ReactComponent as MoonSVG } from './assets/logos/moon.svg'
import { ReactComponent as SunSVG } from './assets/logos/sun.svg'

```

Then style those icons by extending the SVG. The following adds styles to the `svg` tag:

```javascript
const SunIcon = styled(SunSVG)`
  width: 2rem;
  height: 2rem;
  margin-left: auto;
  cursor: pointer;
`

const MoonIcon = styled(MoonSVG)`
  width: 2rem;
  height: 2rem;
  margin-left: auto;
  cursor: pointer;
`

```

Notice how we’re duplicating the code above in both `SunIcon` and `MoonIcon`. Styled Components supports reusing styles.

Reusable CSS in Styled Components
---------------------------------

Styled Components lets us reuse styles using a special helper function named `css`. Go ahead and import it from `styled-components` as follows:

```javascript
import styled, { css, ThemeProvider } from 'styled-components'

```

Now you can reuse styles using the `css` prop as follows:

```javascript
const ThemeIcon = css`
  width: 2rem;
  height: 2rem;
  margin-left: auto;
  cursor: pointer;
`

const MoonIcon = styled(MoonSVG)`
  ${ThemeIcon}
`

const SunIcon = styled(SunSVG)`
  ${ThemeIcon}
`

```

Finally, change the buttons to the following:

```javascript
{theme.type === 'light' && <MoonIcon onClick={setDarkTheme} />}
{theme.type === 'dark' && <SunIcon onClick={setLightTheme} />}

```

The `App.js` file should now like this:

```javascript
import React, { useState } from 'react'
import styled, { css, ThemeProvider } from 'styled-components'
import { ReactComponent as MoonSVG } from './assets/logos/moon.svg'
import { ReactComponent as SunSVG } from './assets/logos/sun.svg'
import { themes } from './utils/themes'

const lightTheme = () => ({
  ...themes['common'],
  ...themes['light'],
})

const darkTheme = () => ({
  ...themes['common'],
  ...themes['dark'],
})

const ThemeIcon = css`
  width: 2rem;
  height: 2rem;
  margin-left: auto;
  cursor: pointer;
`

const MoonIcon = styled(MoonSVG)`
  ${ThemeIcon}
`

const SunIcon = styled(SunSVG)`
  ${ThemeIcon}
`

const App = () => {
    const [theme, setTheme] = useState(lightTheme())
  const setDarkTheme = () => setTheme(darkTheme())
  const setLightTheme = () => setTheme(lightTheme())
    return (
        <ThemeProvider theme={theme}>
            <>
                <Header>Hello React</Header>
                {theme.type === 'light' && <MoonIcon onClick={setDarkTheme} />}
                {theme.type === 'dark' && <SunIcon onClick={setLightTheme} />}
            </>
        </ThemeProvider>
    )
}

export default App

```

And the site should now look like the following. It only shows the `moon` icon on the `light` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956211show-moon-icon-on-light-theme.png)

And it only shows the `sun` icon on the `dark` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956249show-sun-icon-on-dark-theme.png)

Global Styles in Styled Components
----------------------------------

Now go ahead and create a `components/` folder in `src/` directory. Create a file named `Global.js` inside the `components/` folder. Paste the following in it:

```javascript
import { createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
    *, *::before, *::after {
        box-sizing: border-box;
        margin: 0;
        padding: 0;
    }

    html, body {
        height: 100vh;
        max-width: 100%;
        font-size: 62.5%;
        background-color: ${props => props.theme.bgColor};
        color: ${props => props.theme.color};
        font-family: ${props => props.theme.fontFamily};
    }
`

```

We have imported `createGlobalStyle` from `styled-components`. This lets us create global styles. And we later export the `GlobalStyle` variable as a named export. Notice how we’re using themes in our `GlobalStyle`. To make sure theming works, we need to make `GlobalStyle` a child of `ThemeProvider`. Otherwise it won’t work.

Go ahead and import `Global.js` in `App.js` and use it as follows:

```javascript
import { GlobalStyle } from './components/Global'

.
.
.

const App = () => {
    .
    .
    .
    return (
        <ThemeProvider theme={theme}>
            <>
                <GlobalStyle />
                <Header>Hello React</Header>
                {theme.type === 'light' && <MoonIcon onClick={setDarkTheme} />}
                {theme.type === 'dark' && <SunIcon onClick={setLightTheme} />}
            </>
        </ThemeProvider>
    )
}

.
.
.

```

Global styles should be applied to our web app.

Global styles applied on the `light` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956287global-styles-light-theme.png)

Global styles applied on the `dark` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956326global-styles-dark-theme.png)

Now that we know the basics, let’s create the navbar for our Unsplash Redesign. Firstly, go ahead and remove the `Header` styled component from `App.js`.

Then go ahead and add the following in the `App.js` file:

```javascript
import { ReactComponent as SearchSVG } from './assets/logos/search.svg'
import { ReactComponent as UnsplashLogoSVG } from './assets/logos/unsplash-logo.svg'
.
.
.

const Nav = styled.nav`
  display: flex;
  align-items: center;
  background-color: ${props => props.theme.bgColor};
  box-sizing: border-box;
  height: 8.2rem;
  padding-left: 2rem;
  padding-right: 2rem;
  position: fixed;
  width: 100%;
  z-index: 2;
`

const UnsplashLogo = styled(UnsplashLogoSVG)`
  fill: ${props => props.theme.color};
  width: 4rem;
  height: 4rem;
  backface-visibility: hidden;
  box-sizing: border-box;
  overflow: hidden;
  vertical-align: middle;
`

const TitleWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 1rem;
`

const Title = styled.h1`
  display: inline-block;
  font-size: 2rem;
  font-weight: 700;
  color: ${props => props.theme.color};
`

const Subtitle = styled.span`
  font-size: 1.3rem;
  font-weight: 500;
`

const SearchWrapper = styled.div`
  display: flex;
  align-items: center;
  background: ${props => props.theme.searchBgColor};
  border-radius: 3rem;
  border: 0.05rem solid #cbd2d9;
  margin-left: 7rem;
  box-shadow: none;
  :hover {
    border: 0.12rem solid #cbd2d9;
  }
`

const SearchIcon = styled(SearchSVG)`
  fill: #9aa5b1;
  width: 1.6rem;
  height: 1.6rem;
  margin-left: 1.2rem;
  margin-right: 1rem;
`

const SearchBar = styled.input`
  outline: none;
  border: none;
  width: 50rem;
  padding: 1rem 0;
  color: ${props => props.theme.categoryHoverColor};
  background: transparent;
  font-size: 1.4rem;
  font-weight: 300;
`

const App = () => {
  ..
  ..
  ..
  return (
    <ThemeProvider theme={theme}>
      <>
        <GlobalStyle />
        <Nav>
          <UnsplashLogo />
          <TitleWrapper>
            <Title>Unsplash</Title>
            <Subtitle>Photos for Everyone</Subtitle>
          </TitleWrapper>
          <SearchWrapper>
            <SearchIcon />
            <SearchBar placeholder="Search free high-resolution photos" />
          </SearchWrapper>
          {theme.type === 'light' && <MoonIcon onClick={setDarkTheme} />}
          {theme.type === 'dark' && <SunIcon onClick={setLightTheme} />}
        </Nav>
      </>
    </ThemeProvider>
  )
}

..
..
..

```

Notice that the only thing we’ve done differently is to add a `:hover` pseudo class on the `SearchWrapper` component. If you’ve come from Sass or SCSS, then Styled Components is pretty similar. You can even add child selectors inside a Styled Component just like Sass.

The nav should be properly displayed now.

The navbar on the `light` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956358navbar-on-light-theme.png)

The navbar on the `dark` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956388navbar-on-dark-theme.png)

Let’s create a component for listing categories. Create a file `categories.js` in the `utils/` folder and paste the following:

```javascript
export const categories = [
  'Wallpapers',
  'Textures & Patterns',
  'Nature',
  'Current Events',
  'Architecture',
  'Business & Work',
  'Film',
  'Animals',
  'Travel',
  'Fashion',
  'Food & Drink',
  'Spirituality',
  'Experimental',
  'People',
  'Health',
  'Arts & Culture',
]

```

Also, create a component named `CategoriesList.js` in the `components/` folder.

Paste the following code into it:

```javascript
import React from 'react'
import styled from 'styled-components'
import { categories } from '../utils/categories'

const Categories = styled.div`
  padding-bottom: 2rem;
  padding-left: 2rem;
  position: fixed;
  margin-top: 8.2rem;
  top: 0;
  bottom: 0;
  overflow-y: scroll;
  width: 23rem;
  background: ${props => props.theme.bgColor};
  z-index: 2;
`

const CategoryName = styled.a`
  background-color: transparent;
  cursor: pointer;
  box-sizing: border-box;
  color: ${props => props.theme.categoryColor};
  display: block;
  font-size: 1.4rem;
  font-weight: 500;
  line-height: 1;
  padding-bottom: 1.7rem;
  padding-top: 2.1rem;
  text-decoration: none;
  white-space: nowrap;
  :hover {
    color: ${props => props.theme.categoryHoverColor};
    outline: none;
  }
`

const Category = ({ name }) => <CategoryName href="#">{name}</CategoryName>

export const CategoriesList = () => (
  <Categories>
    {categories.map((category, i) => (
      <Category name={category} key={i} i={i} />
    ))}
  </Categories>
)

```

Now go ahead and import it in `App.js` and paste it just below the `Nav` component:

```javascript
import { CategoriesList } from './components/CategoriesList'
..
..
..

const App = () => {
  ..
  ..
  ..
  return (
    <ThemeProvider theme={theme}>
      <>
        <Nav>
          ..
          ..
          ..
        </Nav>
        <CategoriesList />
        ..
        ..
        ..
      </>
    </ThemeProvider>
  )
}

..
..
..

```

Now a list of categories will be shown in the sidebar.

The list of categories on the `light` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956426categories-on-light-theme.png)

The list of categories on the `dark` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956458categories-on-dark-theme.png)

Let’s create our final component that displays pics. Create a file named `pics.js` in the `utils/` folder. Paste the following into it:

```javascript
export const pics = [
  require('../assets/pics/asset 0.jpeg'),
  require('../assets/pics/asset 1.jpeg'),
  require('../assets/pics/asset 2.jpeg'),
  require('../assets/pics/asset 3.jpeg'),
  require('../assets/pics/asset 4.jpeg'),
  require('../assets/pics/asset 5.jpeg'),
  require('../assets/pics/asset 6.jpeg'),
  require('../assets/pics/asset 7.jpeg'),
  require('../assets/pics/asset 8.jpeg'),
  require('../assets/pics/asset 9.jpeg'),
  require('../assets/pics/asset 10.jpeg'),
  require('../assets/pics/asset 11.jpeg'),
  require('../assets/pics/asset 12.jpeg'),
  require('../assets/pics/asset 13.jpeg'),
  require('../assets/pics/asset 14.jpeg'),
  require('../assets/pics/asset 15.jpeg'),
  require('../assets/pics/asset 16.jpeg'),
  require('../assets/pics/asset 17.jpeg'),
  require('../assets/pics/asset 18.jpeg'),
  require('../assets/pics/asset 19.jpeg'),
  require('../assets/pics/asset 20.jpeg'),
  require('../assets/pics/asset 21.jpeg'),
  require('../assets/pics/asset 22.jpeg'),
  require('../assets/pics/asset 23.jpeg'),
  require('../assets/pics/asset 24.jpeg'),
  require('../assets/pics/asset 25.jpeg'),
  require('../assets/pics/asset 26.jpeg'),
  require('../assets/pics/asset 27.jpeg'),
  require('../assets/pics/asset 28.jpeg'),
  require('../assets/pics/asset 29.jpeg'),
  require('../assets/pics/asset 30.jpeg'),
  require('../assets/pics/asset 31.jpeg'),
  require('../assets/pics/asset 32.jpeg'),
  require('../assets/pics/asset 33.jpeg'),
  require('../assets/pics/asset 34.jpeg'),
  require('../assets/pics/asset 35.jpeg'),
  require('../assets/pics/asset 36.jpeg'),
  require('../assets/pics/asset 37.jpeg'),
  require('../assets/pics/asset 38.jpeg'),
  require('../assets/pics/asset 39.jpeg'),
  require('../assets/pics/asset 40.jpeg'),
  require('../assets/pics/asset 41.jpeg'),
  require('../assets/pics/asset 42.jpeg'),
  require('../assets/pics/asset 43.jpeg'),
  require('../assets/pics/asset 44.jpeg'),
  require('../assets/pics/asset 45.jpeg'),
  require('../assets/pics/asset 46.jpeg'),
  require('../assets/pics/asset 47.jpeg'),
  require('../assets/pics/asset 48.jpeg'),
  require('../assets/pics/asset 49.jpeg'),
  require('../assets/pics/asset 50.jpeg'),
  require('../assets/pics/asset 51.jpeg'),
  require('../assets/pics/asset 52.jpeg'),
  require('../assets/pics/asset 53.jpeg'),
]

```

Also create a `Pic.js` file in `components/` folder and paste the following:

```javascript
import React from 'react'
import ImageZoom from 'react-medium-image-zoom'
import styled from 'styled-components'

const ImageWrapper = styled.div`
  margin: 0.5rem;
  transition: all 200ms ease;
  :hover {
    box-shadow: ${props => props.theme.boxShadow};
    transform: scale(1.005);
  }
`

export const Pic = ({ src, i }) => {
  const alt = `Unsplash Redesign #${i}`
  const imageProps = { src, alt }
  return (
    <ImageWrapper>
      <ImageZoom
        image={imageProps}
        zoomImage={imageProps}
        defaultStyles={{
          image: { width: '37rem', height: '48rem' },
        }}
      />
    </ImageWrapper>
  )
}

```

We’re using `react-medium-image-zoom`, which lets you zoom in on an image like [Medium](https://medium.com/).

Now go ahead and import `Pic.js` in `App.js` and paste it just below the `CategoriesList` component. Also, add `Grid` and `GridWrapper` styled components as follows:

```javascript
import { Pic } from './components/Pic'
import { pics } from './utils/pics'

..
..
..

const GridWrapper = styled.div`
  display: flex;
  justify-content: center;
`

const Grid = styled.div`
  padding-top: 8.2rem;
  padding-left: 23rem;
  display: grid;
  grid-template-columns: repeat(3, 38rem);
  margin-bottom: 1.5rem;
`

const App = () => {
  ..
  ..
  ..
  return (
    <ThemeProvider theme={theme}>
      <>
        ..
        ..
        ..
        <CategoriesList />
        <GridWrapper>
          <Grid>
            {pics.map((pic, i) => (
              <Pic src={pic} key={i} i={i} />
            ))}
          </Grid>
        </GridWrapper>
      </>
    </ThemeProvider>
  )
}

..
..
..

```

Now a grid with pics should appear.

The Unsplash redesign with the `light` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956505unsplash-redesign-light-theme.png)

The Unsplash redesign with the `dark` theme:

![](https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2019/08/1566956551unsplash-redesign-dark-theme.png)

We just redesigned Unsplash using `styled-components`. Try zooming in on the image to see it working. You can check the live demo [here](https://unsplash-styled-components.netlify.com/).

Conclusion
----------

Styled Components makes it easy to write plain ol’ CSS in your JS. It allows us to colocate styles with our components so that we can add styles, edit them or remove them easily.

We learned how to add Global Styles, reuse styles using a `css` prop and how to theme an application. We’ve only scratched the surface with `styled-components`; it has a lot more to offer. You can learn more about it on the [official website](https://www.styled-components.com).

We finally redesigned Unsplash using Styled Components. The complete code is available on [GitHub](https://github.com/deadcoder0904/unsplash-styled-components) and the demo is available [here](https://unsplash-styled-components.netlify.com/).

 [![](https://secure.gravatar.com/avatar/e6ef5a14539a2b707c34a8fd392d82c9?s=96&d=mm&r=g)](https://www.sitepoint.com/author/akadam/) 

Meet the author

[Akshay Kadam](https://www.sitepoint.com/author/akadam/)[](https://twitter.com/deadcoder0904)[](https://github.com/deadcoder0904)

Akshay is a creator, computer artist and micropreneur from Mumbai.

[Start a discussion](https://www.sitepoint.com/community/t/how-to-redesign-unsplash-using-styled-components/337269)

Older Posts

*   Stuff we do
    
*   [Premium](/premium/)
*   [Forums](/community/)
*   [Corporate memberships](https://sitepoint.typeform.com/to/fNY7XG)
*   [Become an affiliate](https://sitepoint.tapfiliate.com/)

*   About
    
*   [Our story](/about-us/)
*   [Press room](/press/)

*   Contact
    
*   [Contact us](/contact-us/)
*   [FAQ](https://sitepoint.zendesk.com/hc/en-us)
*   [Publish your book with us](https://sitepoint.typeform.com/to/HtAXVN)
*   [Write an article for us](https://sitepoint.typeform.com/to/DMmYfn)
*   [Advertise](/advertise/)

*   Legals
    
*   [Terms of use](/legals/)
*   [Privacy policy](/privacy-policy/)

*   Connect
    

© 2000 – 2019 SitePoint Pty. Ltd.

![](blob:https://www.sitepoint.com/0c92efaf-c25b-4fa8-8432-2ae74ad4bec9)

FREE Book: JavaScript Best Practice

[Get it Now!](https://www.sitepoint.com/premium/products/Z2lkOi8vbGVhcm5hYmxlL1Byb2R1Y3QvMzAy?utm_source=smartbar&utm_medium=article&utm_campaign=free-book)

Add Tags ?

AddDeleteNo

close

close

[

](#)[

English

](#)[

](#)[

](#)[

](#)

Accessibility Adjustments

*   Virtual Keyboard
    
*   Accessible Mode
    
*   Stop Animations
    

 

Content Adjustments

Font Adjustment

+

0

-

Readable Font

Emphasize Titles

Emphasize Links

Text Magnifier

Content Scaling

+

0

-

Align Center

Adjust Word Spacing

+

0

-

Align Left

Adjust Letter Spacing

+

0

-

Align Right

Adjust Line Height

+

0

-

Align Justify

Color & Display Adjustments

Invert Colors

High Contrast

High Saturation

Dark Contrast

Adjust Text Colors

Cancel

Monochrome

Adjust Title Colors

Cancel

Light Contrast

Adjust Background Colors

Cancel

Navigation Adjustments

Screen Readers Adjustments

Keyboard Navigation

Reading Mode

Hide Images

Mute Sounds

Useful Links

Select an option Home Page Header Footer Main Content

Letter Navigation

Emphasize Focus

Emphasize Mouse Hover

Big Black Cursor

Big White Cursor

Accessible Print

*   Feedback
    
*   Reset Settings
    
*   Statement
    

Accessibility Platform By [accessiBe](https://accessibe.com)

Choose the Interface Language

*   English
    
*   Español
    
*   Deutsche
    
*   Português
    
*   Français
    
*   Italiano
    
*   עברית
    
*   русский
    
*   عربى
    
*   Nederlands
    

Provide Feedback Regarding the Accessibility Interface!

Full Name 

Email Address 

Telephone Number 

Your Feedback

 This is a feedback for people with disabilities to provide feedback on the accessibility level of the website. Any topic that is NOT about the website's accessibility, will not be processed. Accept it by checking the box.

Send Feedback Form

Malkodet  

Thanks for your feedback!

_**Accessibility Statement**_

We firmly believe that the internet should be available and accessible to anyone, and are committed to providing a website that is accessible to the widest possible audience, regardless of circumstance and ability.

To fulfill this promise, we aim to adhere as closely as possible to the Web Content Accessibility Guidelines 2.1 (WCAG 2.1) at the AA level, published by the World Wide Web Consortium (W3C). These guidelines explain how to make web content more accessible to people with a wide array of disabilities. Complying with those guidelines helps us provide a site that is accessible to all people, from the blind to the motor impaired.

This website utilizes various technologies all meant to make it more accessible. We utilize an interface that allows persons with specific disabilities to adjust the website’s UI (user interface) and design it to their personal needs. This interface replaces the older fashioned method of providing all users with the same, lesser but accessible design or user interface.

_**Here are some of the interface’s capabilities:**_

1.  **Font handling -**users can increase and decrease its size, change its family (type), adjust spacing, alignment, line height, and more.
2.  **Color handling - **users can select various color contrast profiles such as light, dark, inverted and monochrome. Additionally, users can swap color schemes of titles, texts, and backgrounds, with over 7 different coloring options.
3.  **Animations – **epileptic users can stop all running animations in an instant, at the click of a button. Those animations include videos, GIFs and CSS flashing transitions.
4.  **Content highlighting – **users can choose to emphasize important elements such as links and titles. They can also choose to highlight focused or hovered elements specifically.
5.  **Audio muting – **users with hearing devices may experience headaches or other issues due to automatic audio playing. This option lets users mute the entire  website instantly.
6.  **Cognitive disorders – **we utilize a search engine that is linked to Wikipedia and Wiktionary, allowing people with cognitive disorders to decipher meanings of phrases, initials, slang, and others.
7.  **Other options**** –** we provide users the option to change cursor color and sizing, use a printing mode, enable a virtual keyboard and much more..

Additionally, the website utilizes an AI-based application that runs in the background and optimizes its accessibility level every 24 hours. This AI engine remediates the website’s HTML issues, as well as functionality and behavior, to the use of blind people that are using screen readers, and the motor impaired that are using a keyboard to operate websites.

_**We utilize AI for the following accessibility components:**_

1.  **Screen-reader optimization:** The AI runs in the background and learns the website’s components top to bottom, thereby providing screen-readers with meaningful data using the ARIA set of attributes. For example, it will provide accurate form labels; descriptions for actionable icons (social media icons, search icons, cart icons, etc.); validation guidance for form inputs; element roles such as buttons, menus, modal dialogues (popups), and others.  
      
    Additionally, the AI scans all of the website's images and provides an accurate and meaningful image-object-recognition-based description as an ALT (alternate text) tag. It will also extract texts that are embedded within the image, using an OCR (optical character recognition) technology. To turn on screen-reader adjustments at any time, click the Alt+1 keyboard combination. Screen-reader users also get automatic announcements to turn adjustments on as soon as they enter the website.  
      
    
2.  **Keyboard navigation optimization:** The AI adjusts the website’s HTML, as well as adding various behaviors using JavaScript code to make the website fully operable by a keyboard. This includes the ability to navigate the entire site using the TAB and SHIFT+TAB keys, operate dropdowns with the arrow keys, close them with ESC, trigger buttons and links using the Enter key, navigate between radio and checkbox elements using the arrow keys, fill them in with the Spacebar or Enter key, and more.  
      
    Additionally, keyboard users will find quick-navigation and content-skip menus (available at any time by clicking Alt+1). The AI will also handle triggered popups by moving the keyboard focus towards them as soon as they appear, and It’ll not let the focus drift outside of it, for best orientation.

_**Browser and Assistive Technology Compatibility:**_

We aim to support the widest array of browsers and assistive technologies as possible, so our users can choose the best fitting tools for them, with as few limitations as possible. Therefore, we have worked very hard to be able to support all the modern, major and popular systems that comprise over 95% of the users’ market share and include: Google Chrome, Mozilla Firefox, Apple Safari, Opera, Microsoft Edge and I.E 11 or above, JAWS and NVDA (screen readers), both for Windows and for MAC users. This website is operable using other systems and platforms as well, though we are not officially supporting them and recommend users to utilize the aforementioned.

_**Notes, Comments, and Feedback:**_

Despite our very best efforts to allow anybody to adjust the website to their needs, it is possible that there will still be pages or sections that are not fully accessible, are in the process of becoming accessible, or are lacking an adequate technological solution to make them accessible. Still, we are constantly improving the accessibility interface, adding, updating and improving its options and features, and developing and adopting new technologies. All this is meant to reach the optimal level of accessibility, at any time and in accordance with technological advancements.

If you’ve discovered a bug or malfunction, if you find any aspect of the interface difficult to use, or if you have improvement ideas, we’ll be happy to hear from you. Simply click on the “Feedback” button at the bottom of the interface, and fill out the feedback form. Alternatively, you can send an email to our web accessibility team using [the following form](#) If you wish to speak with the website owner directly, please contact our team and we’ll provide you with their contact information (we do not publish this information due to data collection and misuse conducted by spambots). 

accessiBe Interface User Guide

accessiBe interface options

The accessiBe interface let's you adjust various parameters of the website to your own needs. All of those, are in order for you to be able the use and view the website to your own optimal level. Amongst the options are changing font family and adjusting sizes, colors and spacing, emphasizing titles and links, adapting the website to screen reading software, keyboard navigation, activating a virtual keyboard and more.

How does it work and how to get started?

accessiBe works in two different methods. The first is manual, which means that you can manually adapt the display and usage of the website to your needs. The second is automatic. Clicking the Basic Accessibility button will automatically generate several processes behind the scenes, that will ease the use of the entire website without you having to adapt anything manually.

I found a bug or malfunction, how can I report it?

In order to leave Feedback or ask for help, simply click the Feedback button at the bottom of the interface. After clicking it, you'll see a popup with a feedback form that will reach the website's owner directly.

How can I adjust the contents of the website?

accessiBe's text adjustment options are the first group of options of the interface. Amongst the options you'll be able to find: Increase or Decrease of the font size, readable font, emphasize titles, emphasize links, text magnifier, letter spacing, word spacing, line height, text alignment to the left, center or the right, and more. Please note that you may not be able to see all the options right there. If you don't, click the 'Show More Text Adjustments' this button will display the entire options.

How can I adjust the display and colors of the website?

accessiBe's color and display options are the second group of options of the interface. Amongst the options you'll be able to find: dark contrast, light contrast, monochrome, show image descriptions, adaptation to screen reading software, stop animations, text color adjustment, title color adjustment, background color adjustment and more. Please note that you may not be able to see all the options right there. If you don't, click the 'Show More Color & Display Adjustments' this button will display the entire options.

How can I adjust the navigational aspects of the website?

accessiBe's Navigational options are the second group of options of the interface. Amongst the options you'll be able to find: emphasize focus, emphasize hover, big white cursor, big black cursor, zoom in and zoom out of the entire website window and more. Please note that you may not be able to see all the options right there. If you don't, click the 'Show More Navigation Adjustments' this button will display the entire options.

How does the keyboard navigation work?

In order to use the keyboard to navigate the website, click the 'Keyboard Navigation' button at the top of the interface. After clicking it, a confirmation window will appear, indicating this feature has been turned on. Afterwards, you'll be able to use the TAB key to navigate between clickable items, SHIFT + TAB to go back, and click ENTER to activate a focused item.

How does the virtual keyboard work?

In order to activate the Virtual Keyboard feature, click the 'Virtual Keyboard' button at the top of the interface. After activating it, the next time you'll hit a form element to enter details, a virtual keyboard will pop on the screen, allowing you the use it instead of a physical keyboard.

How can I skip to main contents?

In order to skip directly to the main content of the website, simply click the 'Skip to Main' button near accessiBe's search feature. Clicking it will scroll you down or up the page directly to the main content of the page.

What is the Basic Accessibility button and what does it do?

The 'Basic Accessibility' button automatically activates a series of events and functions behind the scenes. Those include adapting the website to screen reading software, building image descriptions from context, adapting the website to keyboard navigation, handling non-accessible form elements, and more.

Other interface options

The Accessibility Interface allows you to search for keywords and contents on the page. Entering input to the search field will automatically highlight found matching words, and scroll you down the page directly to them. In addition, at the top of the interface, you'll be able to change the position of the interface (from left to right and the opposite), and to change the size of it (big or small). Clicking the 'eye' icon will allow you to completely hide the accessibility interface.

Hide Accessibility Interface? Please note: If you choose to hide the accessibility interface, you won't be able to see it anymore, unless you clear your browsing history and data. Are you sure that you wish to hide the interface?

Accept Cancel

Continue

Processing the data, please give it a few seconds...

We have noticed that you didn't turn accessibility on. To turn on screen reader adjustments, click Alt+1.

10 Seconds until closing

Playing to screen readers
